### ISSUE TITLE
Python tiledbsoma: repr fails when accessing data on a private bucket

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#315]
This occurs when accessing data on a private bucket.  I have all of the various AWS environment variables set correctly (demonstrated by the successful `atlas.exists()` expression).

It appears that the repr may not be inheriting the context used to open the original SOMA object?

```
In [20]: config
Out[20]: {'py.init_buffer_bytes': 1073741824, 'vfs.s3.region': 'us-west-2'}

In [21]: atlas_uri
Out[21]: 's3://bruce-uswest-tmp/tmp/atlas-20220916/'

In [22]: atlas = soma.SOMACollection(atlas_uri, ctx=tiledb.Ctx(config))

In [23]: atlas.get_uri()
Out[23]: 's3://bruce-uswest-tmp/tmp/atlas-20220916/'

In [24]: atlas.exists()
Out[24]: True

In [25]: atlas
Out[25]: ---------------------------------------------------------------------------
TileDBError                               Traceback (most recent call last)
File ~/projects/TileDB-SingleCell/venv/lib/python3.9/site-packages/IPython/core/formatters.py:707, in PlainTextFormatter.__call__(self, obj)
    700 stream = StringIO()
    701 printer = pretty.RepresentationPrinter(stream, self.verbose,
    702     self.max_width, self.newline,
    703     max_seq_length=self.max_seq_length,
    704     singleton_pprinters=self.singleton_printers,
    705     type_pprinters=self.type_printers,
    706     deferred_pprinters=self.deferred_printers)
--> 707 printer.pretty(obj)
    708 printer.flush()
    709 return stream.getvalue()

File ~/projects/TileDB-SingleCell/venv/lib/python3.9/site-packages/IPython/lib/pretty.py:410, in RepresentationPrinter.pretty(self, obj)
    407                         return meth(obj, self, cycle)
    408                 if cls is not object \
    409                         and callable(cls.__dict__.get('__repr__')):
--> 410                     return _repr_pprint(obj, self, cycle)
    412     return _default_pprint(obj, self, cycle)
    413 finally:

File ~/projects/TileDB-SingleCell/venv/lib/python3.9/site-packages/IPython/lib/pretty.py:778, in _repr_pprint(obj, p, cycle)
    776 """A pprint that just redirects to the normal repr function."""
    777 # Find newlines and replace them with p.break_()
--> 778 output = repr(obj)
    779 lines = output.splitlines()
    780 with p.group():

File ~/projects/TileDB-SingleCell/apis/python/src/tiledbsoma/soma_collection.py:156, in SOMACollection.__repr__(self)
    152 def __repr__(self) -> str:
    153     """
    154     Default display for ``SOMACollection``.
    155     """
--> 156     return "\n".join(self._repr_aux())

File ~/projects/TileDB-SingleCell/apis/python/src/tiledbsoma/soma_collection.py:166, in SOMACollection._repr_aux(self)
    164 lines = [f"{self.get_name()} {self.__class__.__name__}:"]
    165 for key in self.keys():
--> 166     child_lines = self.get(key)._repr_aux()
    167     for line in child_lines:
    168         lines.append("  " + line)

File ~/projects/TileDB-SingleCell/apis/python/src/tiledbsoma/soma_collection.py:82, in SOMACollection.get(self, member_name)
     79     from .factory import _construct_member
     81     member_uri = self._get_child_uri(member_name)
---> 82     self._cached_members[member_name] = _construct_member(
     83         member_name, member_uri, self
     84     )
     86 if member_name in self._cached_members:
     87     return self._cached_members[member_name]

File ~/projects/TileDB-SingleCell/apis/python/src/tiledbsoma/factory.py:41, in _construct_member(member_name, member_uri, parent, ctx)
     38 # Get the class name from TileDB storage. At the TileDB level there are just "arrays" and
     39 # "groups", with separate metadata-getters.
     40 class_name = None
---> 41 object_type = tiledb.object_type(member_uri)
     42 if object_type is None:
     43     raise Exception(f"URI {member_uri} not found")

File tiledb/libtiledb.pyx:5429, in tiledb.libtiledb.object_type()

File tiledb/libtiledb.pyx:579, in tiledb.libtiledb.check_error()

File tiledb/libtiledb.pyx:575, in tiledb.libtiledb._raise_ctx_err()

File tiledb/libtiledb.pyx:560, in tiledb.libtiledb._raise_tiledb_error()

TileDBError: [TileDB::S3] Error: Error while listing with prefix 's3://bruce-uswest-tmp/tmp/atlas-20220916/human/__group/' and delimiter '/'
Exception:  PermanentRedirect
Error message:  Unable to parse ExceptionName: PermanentRedirect Message: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.

```

**Comment by bkmartinjr** [comment]
PR #318 did not resolve this - there is one additional missing place the Ctx needs to be propagated.

```
$ git diff
diff --git a/apis/python/src/tiledbsoma/factory.py b/apis/python/src/tiledbsoma/factory.py
index 33460bc..a6bf39d 100644
--- a/apis/python/src/tiledbsoma/factory.py
+++ b/apis/python/src/tiledbsoma/factory.py
@@ -38,7 +38,7 @@ def _construct_member(
     # Get the class name from TileDB storage. At the TileDB level there are just "arrays" and
     # "groups", with separate metadata-getters.
     class_name = None
-    object_type = tiledb.object_type(member_uri)
+    object_type = tiledb.object_type(member_uri, ctx=ctx)
     if object_type is None:
         raise Exception(f"URI {member_uri} not found")
     elif object_type == "array":
```

**Comment by bkmartinjr** [comment]
@johnkerl - I have confirmed this is fixed by #323, so you can close this bug.
